pyvrp/cpp/search/SwapStar.cpp

```diff
@@ -205,8 +207,11 @@ Cost SwapStar::evaluate(Route *routeU,
 
     Cost deltaCost = static_cast<Cost>(deltaDist);
 
-    // It is not possible to have UAfter == V or VAfter == U, so the positions
-    // are always strictly different
+    // We cannot have that UAfter == V or VAfter == U, as the positions must
+    // always be strictly different.
+    assert(best.VAfter->position != best.U->position);
+    assert(best.UAfter->position != best.V->position);
```

Comment:
These fail at some point. Still not sure why.

Reply:
I think I figured it out. We have a caching mechanism for route ops that we need to update.

Reply:
It's probably a good idea to explicitly test the claim we make here though, so I'm leaving the asserts in.